File: scripts/scripts/testing/automations/incremental/mongolia.R
url <- "https://www1.e-mongolia.mn/covid-19"

data <- rjson::fromJSON(file = "https://e-mongolia.mn/shared-api/api/covid-stat/daily")

date <- data$data$lastDataDate
count <- as.integer(data$data$testedPcrTotal)

add_snapshot(
    count = count,
    date = date,
    sheet_name = "Mongolia",
    country = "Mongolia",
    units = "samples tested",
    source_url = url,
    source_label = "Ministry of Health"
)

File: scripts/scripts/testing/automations/incremental/mongolia.py
from cowidev.utils.web.scraping import request_json
from cowidev.testing.utils.incremental import increment


class Mongolia:
    location = "Mongolia"
    units = "samples tested"
    source_label = "Ministry of Health"
    source_url = "https://e-mongolia.mn/shared-api/api/covid-stat/daily"
    source_url_ref = "https://www1.e-mongolia.mn/covid-19"
    notes = ""

    def read(self):
        data = request_json(self.source_url)
        return {"count": data["data"]["testedPcrTotal"], "date": data["data"]["lastDataDate"]}

    def export(self):
        data = self.read()
        increment(
            count=data["count"],
            sheet_name=self.location,
            country=self.location,
            units=self.units,
            date=data["date"],
            source_url=self.source_url,
            source_label=self.source_label,
        )


def main():
    Mongolia().export()


if __name__ == "__main__":
    main()
